src/main/java/com/example/bookshop/exception/CustomGlobalExceptionHandler.java

```diff
@@ -44,6 +45,15 @@ public ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundExcept
         return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
     }
 
+    @ExceptionHandler(RegistrationException.class)
+    public ResponseEntity<ExceptionDto> handleRegistrationException(RegistrationException ex) {
+        ExceptionDto exceptionDto = new ExceptionDto();
+        exceptionDto.setTimestamp(LocalDateTime.now());
+        exceptionDto.setStatus(HttpStatus.BAD_REQUEST);
+        exceptionDto.setMessage(ex.getMessage());
+        return new ResponseEntity<>(exceptionDto, HttpStatus.BAD_REQUEST);
```

Comment:
Why do you return ExceptionDto dto here, if you return LinkedHashMap in handleEntityNotFoundException method.
Be consistent. Let's reuse code from handleEntityNotFoundException method to handle RegistrationException